plugins/modules/vmware_guest_storage_policy.py

```diff
@@ -0,0 +1,458 @@
+     - '   folder: /datacenter1/vm/folder1'
+     - '   folder: datacenter1/vm/folder1'
+     - '   folder: /folder1/datacenter1/vm'
+     - '   folder: folder1/datacenter1/vm'
+     - '   folder: /folder1/datacenter1/vm/folder2'
+     type: str
+     required: False
+   vm_home:
+     description:
+     - A storage profile policy to set on VM Home.
+     - All values and parameters are case sensitive.
+     - At least one of C(disk) or C(vm_home) are required parameters.
+     required: False
+     type: str
+   disk:
+     description:
+     - A list of disks with storage profile policies to enforce.
+     - All values and parameters are case sensitive.
+     - At least one of C(disk) and C(vm_home) are required parameters.
+     required: False
+     type: list
+     elements: dict
+     suboptions:
+       unit_number:
+         description:
+         - Disk Unit Number.
+         - Valid values range from 0 to 15.
+         type: int
+         required: True
+       policy:
+         description:
+         - Name of the storage profile policy to enforce for the disk.
+         type: str
+         required: True
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Enforce storepol1 policy for disk 0 and 1 using UUID
+  vmware_guest_storage_policy:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    uuid: cefd316c-fc19-45f3-a539-2cd03427a78d
+    disk:
+      - unit_number: 0
+        policy: storepol1
+      - unit_number: 1
+        policy: storepol1
+  delegate_to: localhost
+  register: policy_status
+
+- name: Enforce storepol1 policy for VM Home using name
+  vmware_guest_storage_policy:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    name: hostname1
+    vm_home: storepol1
+  delegate_to: localhost
+'''
+
+RETURN = """
+policy_status:
+    description: Information about changed storage profile policies.
+    returned: always
+    type: dict
+    sample: {
+        "msg": "",
+        "changed_policies": {
+            "disk": [
+                {
+                    "policy": "storepol1",
+                    "unit_number": 0
+                }
+            ],
+            "vm_home": "storepol1"
+        },
+    }
+"""
+
+import traceback
+from ansible.module_utils.basic import missing_required_lib
+PYVMOMI_IMP_ERR = None
+try:
+    from pyVmomi import pbm, vim, VmomiSupport, SoapStubAdapter
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMP_ERR = traceback.format_exc()
+import re
+import ssl
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task
+
+
+class PyVmomiHelper(PyVmomi):
```

Comment:
Could you please use existing [`class SPBM`](https://github.com/ansible-collections/community.vmware/blob/44ac140e30ff0212470b47a29b105d0cdede4df4/plugins/module_utils/vmware_spbm.py#L18)?

Reply:
I'll have a look at this in the coming days.

Reply:
commit ac7a305ece2077e44feecdd7bcd6bb763a448564